src/main/java/io/temporal/internal/sync/DeterministicRunner.java

```diff
@@ -121,4 +122,14 @@ static DeterministicRunner newRunner(
    * called before runUntilAllBlocked.
    */
   void executeInWorkflowThread(String name, Runnable r);
+
+  /**
+   * Creates a new instance of a workflow thread. To be called only from another workflow thread.
+   */
+  WorkflowThread newThread(Runnable runnable, boolean detached, String name);
+
+  /** Only to be called by SyncDecisionContext. */
+  WorkflowThread newThreadNoInterceptor(Runnable runnable, boolean detached, String name);
```

Comment:
I'm trying to understand why this additional API is needed?  Why can't there be a no-op interceptor which does nothing.  It would be good to provide some context as part of API documentation.

Reply:
Refactored to remove the need for newThreadNoInterceptor.